src/error.rs

```diff
@@ -112,7 +56,7 @@ impl From<io::Error> for Error {
 
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{}", self.body_text())
+        write!(f, "{:?}", self)
```

Comment:
Hmm, are you sure this shouldn't instead print the old `body_text()` for synthetic errors?

Reply:
Good call!